presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusClient.java

```diff
@@ -151,10 +162,52 @@ public PrometheusTable getTable(String schema, String tableName)
     private static Map<String, Object> lookupMetrics(URI metadataUri, JsonCodec<Map<String, Object>> metricsCodec)
             throws IOException
     {
-        URL result = metadataUri.toURL();
-        String json = Resources.toString(result, UTF_8);
+        String json = getHttpResponse(metadataUri).string();
         Map<String, Object> metrics = metricsCodec.fromJson(json);
 
         return metrics;
     }
+
+    static ResponseBody getHttpResponse(URI uri)
+            throws IOException
+    {
+        Request.Builder requestBuilder = new Request.Builder();
+        getBearerAuthInfoFromFile().map(bearerToken ->
+                requestBuilder.header("Authorization", "Bearer " + bearerToken));
+        requestBuilder.url(uri.toURL());
+        Request request = requestBuilder.build();
+        Response response = httpClient.newCall(request).execute();
+        if (response.isSuccessful()) {
+            return response.body();
+        }
+        else {
+            throw new PrestoException(REMOTE_TASK_ERROR, "Bad response " + response.code() + response.message());
+        }
+    }
+
+    static Optional<String> getBearerAuthInfoFromFile()
+    {
+        String tokenFileName = config.getBearerTokenFile();
+        if (tokenFileName != null) {
+            DataInputStream reader = null;
```

Comment:
I don't see this being used. Should tokenFIle be getting set to reader or something?

Reply:
Yep, good catch. Changed to use of Guava `Files` which obviated the need for the `DataInputStream` as well as the `finally` to handle closing file. And the ugly `null` checking. The `finally` probably made me scan that as still in use. Fixed in: deccd98